azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageTypeDaoImpl.java

```diff
@@ -44,31 +48,34 @@
   private final DatabaseOperator databaseOperator;
 
   static String INSERT_IMAGE_TYPE =
-      "insert into image_types ( name, description, active, deployable, created_by, modified_by )"
-          + " values (?, ?, ?, ?, ?, ?)";
+      "insert into image_types ( name, description, active, deployable, created_by, "
+          + "created_on, modified_by, modified_on )"
+          + " values (?, ?, ?, ?, ?, ?, ?, ?)";
   static String INSERT_IMAGE_OWNERSHIP =
-      "insert into image_ownerships ( type_id, owner, role, created_by, modified_by ) "
-          + "values (?, ?, ?, ?, ?)";
+      "insert into image_ownerships ( type_id, owner, role, created_by, created_on, modified_by, "
+          + "modified_on ) values (?, ?, ?, ?, ?, ?, ?)";
 
   @Inject
-  public ImageTypeDaoImpl(DatabaseOperator databaseOperator) {
+  public ImageTypeDaoImpl(final DatabaseOperator databaseOperator) {
     this.databaseOperator = databaseOperator;
   }
 
   @Override
-  public int createImageType(ImageType imageType) {
+  public int createImageType(final ImageType imageType) {
     final SQLTransaction<Integer> insertAndGetSpaceId = transOperator -> {
       // insert image type record
-      transOperator.update(INSERT_IMAGE_TYPE, imageType.getName(), imageType.getDescription(),
-          true, imageType.getDeployable().getName(), imageType.getCreatedBy(),
-          imageType.getModifiedBy());
-      int imageTypeId = Long.valueOf(transOperator.getLastInsertId()).intValue();
+      // Passing timestamp from the code base and can be formatted accordingly based on timezone
+      final Timestamp currentTimestamp = Timestamp.valueOf(LocalDateTime.now());
+      transOperator.update(INSERT_IMAGE_TYPE, imageType.getName().toLowerCase(),
+          imageType.getDescription(), true, imageType.getDeployable().getName(),
```

Comment:
The actual enum Deployable i.e. IMAGE, TAR, JAR is used and passed across all the layer. While storing in database enum name rather enum value is stored. And while retrieving from database it again gets converted to Deployable enum. So there is no possibility of any inconsistency and this pattern is followed for all the enums. The actual enum is more for exposing to the user and it should be descriptive enough and the enum value can rather be small or some code. This is also one of the reason for keeping this format and we will have flexibility to change the value going forward.

Reply:
Actually, there is possibility of inconsistency since nothing in the Java syntax prevents developers from start using the name property (lower case values) when referring to the enum instead of using its value (upper case values, the constants) at any given point. 

If I understand correctly the only usage for the `name` property right now will be DB storage because even the users are supposed to use the enum value (IMAGE, TAR, JAR). In that case, why do we need to use the name for DB storage? What prevent us from using the enum value directly?

From the flexibility/extensibility point of view it seems to me that it's the name property, not the enum value what offers text modification without the need to modify the code, so if extensibility is a concern we should probably be using the name for the API fields and the enum value in the code.

Reply:
Actually, there is possibility of inconsistency since nothing in the Java syntax prevents developers from start using the name property (lower case values) when referring to the enum instead of using its value (upper case values, the constants) at any given point.
If I understand correctly the only usage for the name property right now will be DB storage because even the users are supposed to use the enum value (IMAGE, TAR, JAR). In that case, why do we need to use the name for DB storage? What prevent us from using the enum value directly?
From the flexibility/extensibility point of view it seems to me that it's the name property, not the enum value what offers text modification without the need to modify the code, so if extensibility is a concern we should probably be using the name for the API fields and the enum value in the code.